raft-tech/TANF-app
# Issue 762: Logging upload activity

### Summary of Changes
This PR enriches our client logging mechanism by creating LogEntries in Django for our newly Submitted ReportFiles, allowing them to be viewed in Django Admin.

_Addresses issue #762
_Acceptance criteria as stated in the issue_

### How to Test
_List the steps to test the PR_

- Run the frontend and backend apps locally
- Click the Data Files tab
- Click Search
- Add a valid file to any file input
- Click Submit Data Files
- A request should be sent to the backend `/logs` endpoint, and this can be confirmed in the browser's network tab
- In the terminal where the backend is running, the log sent from the frontend should be visible *with the file details of the uploaded file*
- Log into Django Admin
- Ensure a new `Log Entries` table is visible
- Find the Log Entry for the newly submitted upload report
- Ensure Log Entry links to the Report File

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

	  - [x] Demonstrate logs recording information listed above for upload and download (*1)
	  - [x] Demonstrate upload and download activity can be seen in the Django Admin history page for a Report (*2)
	  As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.


1. **Note:** The download work is not captured in this PR and an appropriate logging statement will need to be added later.
2. **Note:** The download work is not captured in this PR but when a logging statement is added in the future, the download activity will be viewable in Django Admin.



### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: 95% (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [x] Backend coverage: 99% (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*


### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*



Does not apply.

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging


### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*


### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

	- None detected
